src/main/java/org/prebid/server/auction/AuctionRequestFactory.java

```diff
@@ -70,12 +70,16 @@ public BidRequest fillImplicitParameters(BidRequest bidRequest, RoutingContext c
         final Device populatedDevice = populateDevice(bidRequest.getDevice(), request);
         final Site populatedSite = bidRequest.getApp() == null ? populateSite(bidRequest.getSite(), request) : null;
         final User populatedUser = populateUser(bidRequest.getUser(), context);
+        final Integer at = bidRequest.getAt();
 
-        if (populatedDevice != null || populatedSite != null || populatedUser != null) {
+        if (populatedDevice != null || populatedSite != null || populatedUser != null || at == null || at == 0) {
             result = bidRequest.toBuilder()
                     .device(populatedDevice != null ? populatedDevice : bidRequest.getDevice())
                     .site(populatedSite != null ? populatedSite : bidRequest.getSite())
                     .user(populatedUser != null ? populatedUser : bidRequest.getUser())
+                    // set the auction type to 1 if it wasn't on the request,
+                    // since header bidding is generally a first-price auction.
+                    .at(at == null || at == 0 ? 1 : at)
```

Comment:
This construct will bite you as it bit me, refer to https://github.com/rubicon-project/prebid-server-java/blob/b984362a449c7d0b882deccf6d47fe92d7842f70/src/main/java/org/prebid/server/auction/AmpRequestFactory.java#L117 for solution